lib/ansible/modules/cloud/vmware/vmware_vm_host_drs_rule.py

```diff
@@ -0,0 +1,467 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: vmware_vm_host_drs_rule
+short_description: Configure VMware DRS rule for virtual machine/s and host system/s in the given cluster
+description:
+- This module can be used to configure VMware DRS rule for virtual machine/s and host system/s in the given cluster.
+version_added: 2.5
+author:
+- "Abhijeet Kasurde (@akasurde)"
+notes:
+- Tested on vSphere 6.5
+requirements:
+- "python >= 2.6"
+- PyVmomi
+options:
+  cluster_name:
+    description:
+    - Desired cluster name where virtual machines and host systems are present for the DRS rule.
+    required: True
+  vms:
+    description:
+    - List of virtual machines name for which DRS rule needs to be applied.
+    - Required if C(state) is set to C(present).
```

Comment:
there's away to have this checked in the declaration of the module:
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/nuage/nuage_vspk.py#L1019-L1020

(same for esxi_hostnames)

Reply:
Done.